HIP/hip-906.md

```diff
@@ -95,13 +93,42 @@ interface IHRC632 {
 }
 ```
 
-Once the above functionality has been implemented in services, an EOA or contract with hbars will be able to call the `hbarAllowance` and `hbarApprove` functions as follows:
+Once the above functionality has been implemented in services, an EOA with hbars will be able to call the `hbarAllowance` and `hbarApprove` functions as follows:
 
 ```
 IHRC(accountAddress).hbarAllowance(address spender)
 IHRC(accountAddress).hbarApprove(address spender, uint256 amount)
 ```
-> The `hbarApprove` function will be callable by either an EOA or a contract as long as the signatures required by the security model are met.
+> The `hbarApprove` function will be callable by an EOA as long as the signatures required by the security model are met.
+
+## Acting on a Contract
+The `IHederaAccountService` system contract will be updated to include the `hbarAllowance` and `hbarApprove` functions.  
+The signature of these calls will be similar to the EOA calls except that they will include the contract address as the first argument.
+
+The following table describes the function selector for the new `hbarAllownace` and `hbarApprove` functions and the associated function signature and response in `IHederaAccountService`.
+
+| Function Selector Hash | Function Signature                                          | Response                                                                               | 
+|------------------------|-------------------------------------------------------------|----------------------------------------------------------------------------------------|
+| 0xfec46666             | hbarAllowance(address owner, address spender)               | (ResponseCode, uint256 - amount of hbar allowances currently available to the spender) | 
+| 0x6e0c21cc             | hbarApprove(address owner, address spender, uint256 amount) | ResponseCode                                                                           |
+
+The solidity interface for `IHederaAccountService` will be the following
+
+```
+interface IHederaAccountService {
+    function hbarAllowance(address owner, address spender) external returns (responseCode, int256);
+    function hbarApprove(address owner, address spender, uint256 amount) external returns (responseCode);
```

Comment:
Now I think of it, why `uint256` and `int256` both used?  E.g., see the `int256` as part of the return in `hbarAllowance`.  

Do we really want `uint256` as an amount of things?  I thought we learned last year that that, with Java, is an issue.  Or was that restricted to int vs uint of 64- or 32- bit things?  (By the time you get to 256 bit things you're in the BigInteger domain anyway.)

Reply:
I'm was the pattern from the token version of the approve call which uses a uint256 to be consistent.  I'll change everything to int256 which seems most correct.